docs/nlu/training-data-format.rst

```diff
@@ -40,6 +40,11 @@ e.g. ``[entity](entity name)``.
     - hey
     - hello
 
+    ## intent: chitchat, response: I am usually called the bank bot, but you could give me a name too. <!-- open domain conversation -->
```

Comment:
```
templates:
  ...
responses:
  chitchat:
    name: ...
```
What's name here?

Reply:
@tmbo  Also the hierarchy would be one level deeper, because there can be multiple types of chitchat(ask weather, ask name, etc.) and each of them have a different response yet the same intent(chitchat)
A detailed document regarding the problems of DX of this is here - https://docs.google.com/document/d/1eblzWgtxWKy5-ClBL3elrs-zch2ai_R8lPd0RgbBctM/edit#heading=h.wsd1z2ow8h8f

Reply:
- I agree with @Ghostvv 's point that we would ideally get people to stop thinking about keys, but I think the *most important* consideration is how intuitive the UX is. 
- Since this feature will likely make it into Rasa X, we should try and keep data schemas as consistent as possible, that would vote in favour of keeping the response utterances in the domain file like the others.
- however, this adds the complication that the NLU training process now needs to know about the `domain.yml` file. So you can't use response selectors if you are using NLU only. Not a dealbreaker but something to consider.

Reply:
I understand the point about all bot utterances being in one place, but I also think that training data should reflect the origin of the utterance, which is in this case not the core.
I think in general domain is not a place for text templates, because every time they change the bot shouldn't be retrained, however, if response answer is changed, it should

Reply:
- Extending on @amn41 's point about domain needed inside NLU process, should we in parallel assess how tedious/useful that is? Or maybe that's out of scope in this discussion? I think going forward there might be a lot of components which do not fall neatly under just NLU or Core, like this one. Here is the issue for this - https://github.com/RasaHQ/rasa/issues/4285
- @Ghostvv Since we are not supporting multiple responses for now, if response phrases are placed inside a separate section of domain, it should be fine isn't it? Because if they are changed the component does need to get retrained. However in that case, the core model would also get retrained. Can we prevent just that?

Reply:
I've checked the "domain in NLU training feasability": code wise it is a three line change (thanks to @wochinge great work on data importers). But as @amn41 it has certain implications:
- training of NLU models containing open domain classifiers don't work without a domain
- hence it might get a bit messy for the `rasa train nlu` cli

In my opinion, NLU only is non-focus. In this case, if you'd really want you'd still be able to use it providing a stripped-down domain only containing the needed responses.

Regarding checks on which part needs to be retrained: we should keep this in mind but this should not pose an issue.

lets continue discussion on the shared doc then